src/test/unit_tests/test_notifications.js

```diff
@@ -0,0 +1,203 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+// disabling init_rand_seed as it takes longer than the actual test execution
+process.env.DISABLE_INIT_RANDOM_SEED = "true";
+
+const fs = require('fs');
+const path = require('path');
+
+const config = require('../../../config');
+
+// setup coretest first to prepare the env
+const { get_coretest_path, generate_s3_client, TMP_PATH } = require('../system_tests/test_utils');
+const coretest_path = get_coretest_path();
+const coretest = require(coretest_path);
+
+const { rpc_client, EMAIL, POOL_LIST } = coretest;
+
+console.log('POOL_LIST =', POOL_LIST);
+
+const tmp_path = path.join(TMP_PATH, 'test_notifications');
+//dir for connect files
+const tmp_connect = path.join(tmp_path, 'connect');
+//dir for notification persistent files
+const notif_logs_path = path.join(tmp_path, 'notif_logs');
+config.NOTIFICATION_LOG_DIR = notif_logs_path;
+
+coretest.setup({pools_to_create: [POOL_LIST[1]]});
+const mocha = require('mocha');
+const http = require('http');
+const assert = require('assert');
+const timer = require('node:timers/promises');
+const notifications_util = require('../../util/notifications_util');
+
+const http_connect_filename = 'http_connect.json';
+const http_connect_path = path.join(tmp_connect, http_connect_filename);
+//content of connect file, will be written to a file in before()
+const http_connect = {
+    agent_request_object: {"host": "localhost", "port": 9999, "timeout": 1500},
+    request_options_object: {"auth": "amit:passw", "timeout": 1500},
+    notification_protocol: 'http',
+    name: 'http_notif'
+};
+
+const Bucket = 'notif';
+
+config.NOTIFICATION_CONNECT_DIR = tmp_connect;
+
+let http_server = null;
+let server_done = false;
```

Comment:
Please add comments on those variables (so we can easy understand their exact use).